src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexingException.cs

```diff
@@ -3,17 +3,68 @@
 
 using System;
 using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Serialization;
 
 namespace Microsoft.Azure.WebJobs.Host.Indexers
 {
-    [SuppressMessage("Microsoft.Design", "CA1064:ExceptionsShouldBePublic")]
+    /// <summary>
+    /// Exception that occurs when the <see cref="JobHost"/> encounters errors when trying
+    /// to index job methods on startup.
+    /// </summary>
     [SuppressMessage("Microsoft.Design", "CA1032:ImplementStandardExceptionConstructors")]
     [Serializable]
-    internal class FunctionIndexingException : Exception
+    public class FunctionIndexingException : Exception
     {
+        /// <summary>
+        /// Constructs a new instance.
+        /// </summary>
+        /// <param name="methodName">The name of the method in error.</param>
+        /// <param name="innerException">The inner exception.</param>
         public FunctionIndexingException(string methodName, Exception innerException)
             : base("Error indexing method '" + methodName + "'", innerException)
         {
+            MethodName = methodName;
+        }
+
+        /// <summary>
+        /// Constructs a new instance.
+        /// </summary>
+        /// <param name="info">The <see cref="SerializationInfo"/>.</param>
+        /// <param name="context">The <see cref="StreamingContext"/>.</param>
+        protected FunctionIndexingException(SerializationInfo info, StreamingContext context) : base(info, context)
+        {
+            if (info == null)
+            {
+                throw new ArgumentNullException("info");
+            }
```

Comment:
I'm open to relaxing rules where they make sense. In some repos where the APIs aren't a public SDK it makes sense, but this repo IS a public repo, so I'm more cautious. Let's discuss another time.